e2e/alertmanager_test.go

```diff
@@ -135,6 +137,27 @@ func testAlertmanagerDeployed(ctx context.Context, t *OperatorContext) {
 			return false, fmt.Errorf("unexpected annotations (-want, +got): %s", diff)
 		}
 
+		// If spec is empty, no need to assert EXTRA_ARGS.
+		if spec == nil {
+			return true, nil
+		}
+		var wantArgs []string
+		for _, c := range ss.Spec.Template.Spec.Containers {
+			if c.Name != "alertmanager" {
+				continue
+			}
+			// We're mainly interested in the dynamic flags but checking the entire set including
+			// the static ones is ultimately simpler.
+			if externalURL := spec.ExternalURL; externalURL != "" {
```

Comment:
Bit confused again - can't we prepare wantArgs before looping through Alertmanager statetful set containers? It feel like you do it inside this loop for a reason, but I can't find any 🤔